### PR TITLE
CLOUDSTACK-9244 Fix setting up RFC1918 routes

### PR DESCRIPTION
**Pull request by remibergsma**
Public ip var was empty so routes were always set. Corrected this.

### CODE REVIEW

**Comment by nislim** [comment]
Fixed my problem, my routing tables are now looking as I expected.
Thanks

File: systemvm/patches/debian/config/etc/init.d/cloud-early-config [hunk 1]
```diff
@@ -1166,6 +1166,9 @@ setup_storage_network() {
 }
 
 setup_system_rfc1918_internal() {
+  public_ip=$ETH2_IP
+  [ "$ETH2_IP" == "0.0.0.0" ] && public_ip=$ETH1_IP
```
Comment (DaanHoogland) [inline systemvm/patches/debian/config/etc/init.d/cloud-early-config]: shouldn't line 1195 begone, then? I think it does the same

**Comment by DaanHoogland** [comment]
If this helps @nislim And is tested it should go in.
I feel uncertain of this functionality. It is called from sestro and console proxy setup functions but the code is also present in those functions. I think the lines 1195 and 1252 should be removed and the functionality abstracted out.

**Comment by remibergsma** [comment]
@DaanHoogland I think we agree cloud-early-config needs major refactoring as it is one big copy/paste thing. Just don't think this is the time to do it.

All this does is pickup the public ip address and use it, like other parts also do.

**Comment by remibergsma** [comment]
Discussed with @DaanHoogland and he sent me a patch ;-)

**Comment by remibergsma** [comment]
@nislim Can you please test the enhanced patch?

**Comment by remibergsma** [comment]
`cloud-early-config` now logs:

```
Not setting up route of RFC1918 space to 192.168.22.1 befause 192.168.23.2 is RFC1918..
```

Routes are not set as expected:
```
root@v-85-VM: # route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         192.168.23.1    0.0.0.0         UG    0      0        0 eth2
169.254.0.0     0.0.0.0         255.255.0.0     U     0      0        0 eth0
192.168.22.0    0.0.0.0         255.255.255.0   U     0      0        0 eth1
192.168.23.0    0.0.0.0         255.255.255.0   U     0      0        0 eth2
```

And when it has a real public ip a route is set to the internal gw. That functionality still works and didn't change.

```
Setting up route of RFC1918 space to 192.168.22.1 
```

Routes are setup now:
```
root@v-1357-VM: # route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         x.y.227.1       0.0.0.0         UG    0      0        0 eth2
10.0.0.0        192.168.22.1    255.0.0.0       UG    0      0        0 eth1
169.254.0.0     0.0.0.0         255.255.0.0     U     0      0        0 eth0
172.16.0.0      192.168.22.1    255.240.0.0     UG    0      0        0 eth1
192.168.0.0     192.168.22.1    255.255.0.0     UG    0      0        0 eth1
192.168.22.0    0.0.0.0         255.255.255.0   U     0      0        0 eth1
 x.y.227.0      0.0.0.0         255.255.255.0   U     0      0        0 eth2
```

**Comment by DaanHoogland** [comment]
looks good

**Comment by nislim** [comment]
Same result with the refactored code:
```
root@v-23-VM:~# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.0.12.254     0.0.0.0         UG    0      0        0 eth2
10.0.10.0       0.0.0.0         255.255.255.0   U     0      0        0 eth1
10.0.12.0       0.0.0.0         255.255.255.0   U     0      0        0 eth2
169.254.0.0     0.0.0.0         255.255.0.0     U     0      0        0 eth0
```
Code looks good to me

**Pull request merged**